server/src/main/java/io/crate/execution/engine/collect/ShardCollectorProvider.java

```diff
@@ -119,13 +119,16 @@ public ShardRowContext shardRowContext() {
         assert collectPhase.maxRowGranularity() == RowGranularity.DOC :
             "granularity must be DOC";
 
+        boolean isClosedIndex = indexShard.mapperService() == null;
```

Comment:
```suggestion
        boolean isOpenIndex = indexShard.mapperService() != null;
```

This could improve readability as all checks on the boolean are otherwise negated by `!isClosedIndex`.